src/interface/variable_pack.hpp

```diff
@@ -70,44 +139,62 @@ template <typename T>
 class VariablePack {
  public:
   VariablePack() = default;
+
   VariablePack(const ViewOfParArrays<T> &view, const ParArray1D<int> &sparse_ids,
                const ParArray1D<int> &vector_component, const std::array<int, 4> &dims)
       : v_(view), sparse_ids_(sparse_ids), vector_component_(vector_component),
         dims_(dims), ndim_((dims[2] > 1 ? 3 : (dims[1] > 1 ? 2 : 1))) {}
+
   KOKKOS_FORCEINLINE_FUNCTION
-  ParArray3D<T> &operator()(const int n) const { return v_(n); }
+  bool IsAllocated(const int n) const {
+    assert(0 <= n && n < dims_[3]);
+    return v_(n).size() > 0;
+  }
+
+  KOKKOS_FORCEINLINE_FUNCTION
+  ParArray3D<T> &operator()(const int n) const {
+    assert(IsAllocated(n));
+    return v_(n);
+  }
+
   KOKKOS_FORCEINLINE_FUNCTION
   T &operator()(const int n, const int k, const int j, const int i) const {
+    assert(IsAllocated(n));
     return v_(n)(k, j, i);
   }
+
+  // TODO(JL) This is here so code templated on VariablePack and MeshBlockPack doesn't
+  // need to change, but maybe we should add assert(m == 0)?
   KOKKOS_FORCEINLINE_FUNCTION
   T &operator()(const int m, const int n, const int k, const int j, const int i) const {
+    assert(IsAllocated(n));
     return v_(n)(k, j, i);
   }
+
   KOKKOS_FORCEINLINE_FUNCTION
   int GetDim(const int i) const {
     assert(i > 0 && i < 6);
     return (i == 5 ? 1 : dims_[i - 1]);
   }
+
   KOKKOS_FORCEINLINE_FUNCTION
-  int GetSparse(const int n) const {
+  int GetSparseID(const int n) const {
     assert(0 <= n && n < dims_[3]);
     return sparse_ids_(n);
   }
+
   KOKKOS_FORCEINLINE_FUNCTION
-  int GetSparseId(const int n) const { return GetSparse(n); }
-  KOKKOS_FORCEINLINE_FUNCTION
-  int GetSparseIndex(const int n) const { return GetSparse(n); }
-  KOKKOS_FORCEINLINE_FUNCTION
-  bool IsVector(const int n) const {
-    assert(0 <= n && n < dims_[3]);
-    return vector_component_(n) != NODIR;
-  }
```

Comment:
Both of these functions returned the same thing, namely `GetSparse(n);` So we had `GetSparse(n) == GetSparseId(n) == GetSparseIndex(n)` for all valid `n`. So if they were meant to be different, this was a bug.

If it wasn't a bug, then I'd say we should only have one function instead of 3 aliases that return exactly the same thing.

Reply:
They were the same now, because we didn't have dense on block, but they were intended to be different when dense on block came online.

Reply:
sure that sounds good. Can you leave the duplicate function in place until then?

Reply:
Certainly